### PR TITLE
Remove ResponseRenderer::callExit() calls from auth plugins

### PR DESCRIPTION
**Pull request by MauricioFauth**
Refactors authentication plugins to return a response object instead of throwing `ExitException`.

### CODE REVIEW

File: src/Exceptions/AuthenticationFailure.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+namespace PhpMyAdmin\Exceptions;
+
+use RuntimeException;
+use Throwable;
+
+use function __;
+
+final class AuthenticationFailure extends RuntimeException
+{
+    public const SERVER_DENIED = 'server-denied';
```
Comment (kamil-tekiela) [inline src/Exceptions/AuthenticationFailure.php:14]: Is this a good candidate for an enum?

File: src/Exceptions/AuthenticationFailure.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+     */
+    public static function rootDenied(): self
+    {
+        return new self(self::ROOT_DENIED, __('Access denied!'));
+    }
+
+    /**
+     * Empty password is denied
+     */
+    public static function emptyDenied(): self
```
Comment (kamil-tekiela) [inline src/Exceptions/AuthenticationFailure.php:57]:
```suggestion
    public static function emptyPasswordDenied(): self
```

File: src/Exceptions/AuthenticationFailure.php [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+     */
+    public static function allowDenied(): self
+    {
+        return new self(self::ALLOW_DENIED, __('Access denied!'));
+    }
+
+    /**
+     * User 'root' is denied in configuration
+     */
+    public static function rootDenied(): self
```
Comment (kamil-tekiela) [inline src/Exceptions/AuthenticationFailure.php:49]:
```suggestion
    public static function rootDeniedByConfiguration(): self
```
Can the comment be merged into the method name? Does it have to be this name?

File: src/Exceptions/AuthenticationFailure.php [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        return new self(
+            self::EMPTY_DENIED,
+            __('Login without a password is forbidden by configuration (see AllowNoPassword).'),
+        );
+    }
+
+    /**
+     * Automatically logged out due to inactivity
+     */
+    public static function noActivity(): self
```
Comment (kamil-tekiela) [inline src/Exceptions/AuthenticationFailure.php:68]:
```suggestion
    public static function loggedOutDueToInactivity(): self
```

File: src/Exceptions/AuthenticationFailure.php (hunk 1 above)
Comment (MauricioFauth) [reply src/Exceptions/AuthenticationFailure.php:14]: I thought about it, but a custom auth plugin may want to add its own failure type.

**Pull request merged**